vm/src/obj/objsequence.rs

```diff
@@ -137,7 +139,154 @@ impl<T: Clone> PySliceableSequence for Vec<T> {
     }
 }
 
-pub fn get_item(
+pub trait SequenceProtocol
+where
+    Self: Sized,
+{
+    fn get_elements(&self) -> Vec<PyObjectRef>;
```

Comment:
we definitely don't want to copy the entire vec just to eg check its size.

Reply:
Agree. I can't do a `Deref` inside the trait. Do you have a suggestion for something nicer then returning the `Refcell`?

Reply:
You could return an `&[PyObjectRef]`, or if that wouldn't work an `impl Deref<Target = [PyObjectRef]>`.